Sources/StreamChat/Models/Attachments/AnyAttachmentPayload.swift

```diff
@@ -6,9 +6,22 @@ import Foundation
 
 /// A protocol an attachment payload type has to conform in order it can be
 /// attached to/exposed on the message.
-public protocol AttachmentPayload: Codable {
+public protocol AttachmentPayload: Codable, Equatable {
     /// A type of resulting attachment.
     static var type: AttachmentType { get }
+
+    /// Update the url of the attachment with the remote one.
+    /// This function will be called once an attachment has successfully uploaded.
+    /// - Parameter url: The remote url of the attachment.
+    mutating func updateRemoteUrl(_ url: URL)
```

Comment:
The method name's a bit confusing - it doesn't update the remote url, but the attachment url with the remote one.
Maybe sth like this would be better: `updateAttachmentUrl(_ remoteUrl: URL)`?